src/Features/LanguageServer/Protocol/Handler/RequestExecutionQueue.QueueItem.cs

```diff
@@ -82,15 +88,18 @@ public async Task CallbackAsync(RequestContext context, CancellationToken cancel
                 try
                 {
                     await _callbackAsync(context, cancellationToken).ConfigureAwait(false);
+                    Metrics.RecordSuccess();
                 }
                 catch (OperationCanceledException)
                 {
                     _logger.TraceInformation($"{MethodName} - Canceled");
+                    Metrics.RecordCancellation();
                     throw;
                 }
                 catch (Exception ex)
                 {
                     _logger.TraceException(ex);
+                    Metrics.RecordFailure();
```

Comment:
are these associated with any sort of request?  taht way we can see: request X is commonly failing.

Reply:
these looks like static calls.  if theyr'e not, can you use `this.Metrics.`

Reply:
Yes, all the request time metrics are associated with the LSP method name.  I didn't associate the time in queue metrics with a method though, as the time in queue shouldn't really vary based on the method.

Reply:
well, i would like to know if a user is submitting a completion call, and it takes 15 seconds, 14 of which are in queue, right?  I get that time in queue is bad for everyone.  but i'd still like to accurately measure *per* request how logn the total time is on our end.

Reply:
I think I am doing what you're saying unless I'm misunderstanding - basically I record two things

1.  Time in queue - this is how long a request sits in the queue.  I do not output this metric per method, as the time in queue isn't affected by method name.
2.  Total request time - I output this metric per method since the execution time will vary depending on the method being called.  This is the overall time to handle the request, from when we got it to when we respond (including time in queue).